### PR TITLE
Add reindexing service

### PR DESCRIPTION
**Pull request by acoburn**
See: https://jira.duraspace.org/browse/FCREPO-1488

This creates a `fcrepo-reindexing` module in fcrepo-camel-toolbox

### CODE REVIEW

**Comment by awoods** [comment]
I assume the application.properties also needs to be updated:
https://github.com/fcrepo4-labs/fcrepo-camel-toolbox/blob/master/fcrepo-camel-webapp/src/main/resources/application.properties

**Comment by acoburn** [comment]
Yes, they do. Thanks for catching that!

File: fcrepo-indexing-solr/pom.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
       <groupId>org.apache.camel</groupId>
       <artifactId>camel-http4</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>org.apache.camel</groupId>
```
Comment (awoods) [inline fcrepo-indexing-solr/pom.xml]: I am not seeing where `camel-restlet` is used in `fcrepo-indexing-solr`.
I removed this dependency locally, and the build completes successfully.
If there is no particular reason for this dependency here, let's drop it.

Comment (acoburn) [inline fcrepo-indexing-solr/pom.xml]: Nope, that should be removed. That's from when I started implementing the reindexing piece inside the indexing-solr application. I have a number of changes that are part of a forthcoming commit.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4-labs/fcrepo-camel-toolbox/commit/170c289d3dd6d8064d90defae81d60fba5f53dfb

**Pull request closed without merging**